### ISSUE TITLE
use multiple cores to train LGBMClassifier pipeline

### ISSUE THREAD

**Issue by chaupmcs** [jpmml/sklearn2pmml#131]
Hi @vruusmann,

I'm wondering if I can speed up the training process with multiple-core machine?

The code is as following:

    params = {'classifier__categorical_feature':[1, 2, 3, 4, 6, 7] }
    classifier = lgb.LGBMClassifier(n_estimators=5, learning_rate=0.1, num_leaves=10, max_depth=2, n_jobs=20)
    pipeline = PMMLPipeline([
       ("mapper", mapper),
       ("classifier", classifier)
      ])

    pipeline.fit(X = X_train, y = y_train, **params)

The code above works just fine, but it doesn't take advantage of a plenty of cores in my server even when I specify `n_jobs = 20`. It takes me hours to run. In case of "normal" package, I can shorten the training time by increasing the number of cores.
Is it possible with `pipeline`.
Thanks a lot.

**Comment by vruusmann** [comment]
The SkLearn2PMML package is not the limiting factor here - class `sklearn2pmml.pipeline.PMMLPipeline` is a "thin" subclass of `sklearn.pipeline.Pipeline`.

To see it yourself, if you replace `PMMLPipeline` with `Pipeline` in your code example, then the execution will still be single-thread/single-core.

The right place to ask this question is either at LightGBM project (if you think that `LGBMClassifier` is the limiter), or at the Scikit-Learn project (if you think that `Pipeline` is the limiter).

Can't help, so closing this issue.
